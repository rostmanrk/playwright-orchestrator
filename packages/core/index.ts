export * from './src/types/adapters';
export * from './src/types/test-info';
