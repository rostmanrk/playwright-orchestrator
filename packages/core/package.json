{
    "name": "@playwright-orchestrator/core",
    "version": "1.1.3",
    "description": "Core lib and cli for Playwright test orchestration",
    "keywords": [
        "playwright",
        "testing",
        "automation",
        "orchestration",
        "e2e"
    ],
    "author": "Rostyslav Kudrevatykh",
    "license": "Apache-2.0",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/rostmanrk/playwright-orchestrator.git"
    },
    "module": "dist/index.js",
    "types": "dist/index.d.ts",
    "bin": {
        "playwright-orchestrator": "./dist/cli.js"
    },
    "files": [
        "dist"
    ],
    "dependencies": {
        "@commander-js/extra-typings": "^13.0.0",
        "ansi-escapes": "^7.0.0",
        "chalk": "^5.4.1",
        "commander": "^13.0.0",
        "boxen": "^8.0.1",
        "typescript": "^5.0.0",
        "uuid": "^11.0.3"
    },
    "devDependencies": {},
    "type": "module",
    "main": "dist/index.js",
    "exports": {
        ".": {
            "types": "./dist/index.d.ts",
            "default": "./dist/index.js"
        },
        "./tests-info-reporter": {
            "import": "./dist/tests-info-reporter.js",
            "require": "./dist/tests-info-reporter.js"
        }
    },
    "peerDependencies": {
        "@playwright/test": "^1.44.0",
        "@playwright-orchestrator/file": "^1.0.0",
        "@playwright-orchestrator/dynamo-db": "^1.0.0",
        "@playwright-orchestrator/pg": "^1.0.0",
        "@playwright-orchestrator/mysql": "^1.0.0",
        "@playwright-orchestrator/mongo": "^1.0.0"
    },
    "peerDependenciesMeta": {
        "@playwright-orchestrator/file": {
            "optional": true
        },
        "@playwright-orchestrator/dynamo-db": {
            "optional": true
        },
        "@playwright-orchestrator/pg": {
            "optional": true
        },
        "@playwright-orchestrator/mysql": {
            "optional": true
        },
        "@playwright-orchestrator/mongo": {
            "optional": true
        }
    },
    "scripts": {
        "prepare": "cp ../../README.md ./ && cp ../../LICENSE.md ./"
    }
}
