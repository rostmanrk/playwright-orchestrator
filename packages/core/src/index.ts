export * from './types/adapters.js';
export * from './types/reporter.js';
export * from './types/test-info.js';
