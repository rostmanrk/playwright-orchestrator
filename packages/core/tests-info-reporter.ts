import { TestsInfoReporter } from './src/playwright-tools/tests-info-reporter';
export default TestsInfoReporter;
