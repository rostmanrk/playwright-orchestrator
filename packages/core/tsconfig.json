{
    "extends": "../../tsconfig.base.json",
    "compilerOptions": {
        "outDir": "./dist",
        "rootDir": "./src"
    },
    "include": ["./src/**/*.ts", "src/playwright-tools/annotations.cts"],
    "exclude": ["dist", "package.json"]
}
