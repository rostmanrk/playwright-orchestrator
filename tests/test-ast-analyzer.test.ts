import { test, expect } from 'vitest';
import { promisify } from 'node:util';
import child_process from 'node:child_process';
import { TestRunInfo } from '@playwright-orchestrator/core';
import path from 'node:path';

const exec = promisify(child_process.exec);

test('test custom info reporter', async () => {
    const commandResult = await exec(
        'npx playwright test --list --reporter "@playwright-orchestrator/core/tests-info-reporter" -j 2',
    );
    const testRunInfo = JSON.parse(commandResult.stdout) as TestRunInfo;
    /// Make the outputDir relative to the current working directory to make the snapshot stable.
    if (testRunInfo.config.configFile) {
        testRunInfo.config.configFile = path.relative(process.cwd(), testRunInfo.config.configFile);
    }
    for (const project of testRunInfo.config.projects) {
        project.outputDir = path.relative(process.cwd(), project.outputDir);
    }
    expect(testRunInfo).toMatchSnapshot();
}, 60000);
