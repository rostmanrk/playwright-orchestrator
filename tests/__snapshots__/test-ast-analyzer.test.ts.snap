// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test custom info reporter 1`] = `
{
  "config": {
    "configFile": "playwright.config.ts",
    "projects": [
      {
        "name": "chromium",
        "outputDir": "test-results/chromium",
      },
      {
        "name": "firefox",
        "outputDir": "test-results",
      },
      {
        "name": "webkit",
        "outputDir": "test-results",
      },
    ],
    "workers": 2,
  },
  "testRun": {
    "basic-case.spec.ts": {
      "10:9": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 30000,
      },
      "4:5": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 30000,
      },
    },
    "constant-timeout.spec.ts": {
      "12:9": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 45000,
      },
      "16:9": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 50000,
      },
      "23:13": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 135000,
      },
      "4:5": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 40000,
      },
    },
    "failed.spec.ts": {
      "3:5": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 30000,
      },
    },
    "serial-top-level.spec.ts": {
      "0:0": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 60000,
      },
    },
    "serial.spec.ts": {
      "10:9": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 30000,
      },
      "14:10": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 85000,
      },
      "4:5": {
        "projects": [
          "chromium",
          "firefox",
          "webkit",
        ],
        "timeout": 30000,
      },
    },
  },
}
`;
